lib/sprockets/sass/v2/importer.rb

```diff
@@ -0,0 +1,205 @@
+          return nil if engine_imports.empty?
+          ::Sass::Engine.new engine_imports, options.merge(
+            filename: base_path.to_s,
+            syntax: syntax(base_path.to_s),
+            importer: self,
+            custom: { sprockets_context: context }
+          )
+        end
+
+        # Finds an asset from the given path. This is where
+        # we make Sprockets behave like Sass, and import partial
+        # style paths.
+        def resolve(context, path, base_path)
+          paths, _root_path = possible_files(context, path, base_path)
+          paths.each do |file|
+            context.resolve(file.to_s) do |found|
+              return found if context.asset_requirable?(found)
+            end
+          end
+          nil
+        end
+
+        # Finds all of the assets using the given glob.
+        def resolve_glob(context, glob, base_path)
+          base_path      = Pathname.new(base_path)
+          path_with_glob = base_path.dirname.join(glob).to_s
+
+          Pathname.glob(path_with_glob).sort.select do |path|
+            asset_requirable = context.asset_requirable?(path)
+            path != context.pathname && asset_requirable
+          end
+        end
+
+        # Returns all of the possible paths (including partial variations)
+        # to attempt to resolve with the given path.
+        def possible_files(context, path, base_path)
+          path      = Pathname.new(path)
+          base_path = Pathname.new(base_path).dirname
+          partial_path = partialize_path(path)
+          additional_paths = [Pathname.new("#{path}.css"), Pathname.new("#{partial_path}.css"), Pathname.new("#{path}.css.#{syntax(path)}"), Pathname.new("#{partial_path}.css.#{syntax(path)}")]
+          paths = additional_paths.concat([path, partial_path])
+
+          # Find base_path's root
+          paths, root_path = add_root_to_possible_files(context, base_path, path, paths)
+          [paths.compact, root_path]
+        end
+
+        def add_root_to_possible_files(context, base_path, path, paths)
+          env_root_paths = context.environment.paths.map { |p| Pathname.new(p) }
+          root_path = env_root_paths.find do |env_root_path|
+            base_path.to_s.start_with?(env_root_path.to_s)
+          end
+          root_path ||= Pathname.new(context.root_path)
+          # Add the relative path from the root, if necessary
+          if path.relative? && base_path != root_path
+            relative_path = base_path.relative_path_from(root_path).join path
+            paths.unshift(relative_path, partialize_path(relative_path))
+          end
+          [paths, root_path]
+        end
+
+        # Returns the partialized version of the given path.
+        # Returns nil if the path is already to a partial.
+        def partialize_path(path)
+          return unless path.basename.to_s !~ /\A_/
+          Pathname.new path.to_s.sub(/([^\/]+)\Z/, '_\1')
+        end
+
+        def opposite_syntax(path)
```

Comment:
I was referring to the `opposite_syntax` method - I don't see where that's being used